### PR TITLE
GOV-517 Deployment file fix for Identity Account Mapper

### PR DESCRIPTION
**Pull request by apurbraj**
## Description

* Deployment file fix for identity account mapper. [GOV-517](https://mifosforge.jira.com/browse/GOV-517?atlOrigin=eyJpIjoiZmNhYzYyNWY5MDYzNDhjOWEzYTI2Y2U2ZGVlNjBkZTkiLCJwIjoiaiJ9)
* Add a link to teh design document or include the design bullet points related to this PR here.

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

[GOV-517]: https://mifosforge.jira.com/browse/GOV-517?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helm/g2p-sandbox/requirement.yaml
Comment (fynmanoj) [inline helm/g2p-sandbox/requirement.yaml:1]: remove these changes and add them to chart.yaml as dependency.

File: helm/g2p-sandbox/values.yaml [hunk 1]
```diff
@@ -136,10 +136,12 @@ ph-ee-engine:
         CREATE DATABASE messagegateway;
         CREATE DATABASE `rhino`;
         CREATE DATABASE `gorilla`;
+        CREATE DATABASE `IdentityAccountMapper`;
```
Comment (fynmanoj) [inline helm/g2p-sandbox/values.yaml:139]: use  lowercase name for database

File: helm/g2p-sandbox/values.yaml [hunk 2]
```diff
@@ -490,3 +492,51 @@ ph-ee-engine:
       ingress:
         enabled: false
         hostname: admin-kong.sandbox.mifos.io
+
+  account_mapper:
+    enabled: true
+    image: "419830066942.dkr.ecr.ap-south-1.amazonaws.com/ph-ee-vouchers"
```
Comment (fynmanoj) [inline helm/g2p-sandbox/values.yaml:498]: please point to the docker hub images
why the voucher image is used here in account mapper?

File: helm/g2p-sandbox/values.yaml [hunk 3]
```diff
@@ -490,3 +492,51 @@ ph-ee-engine:
...
+    imageTag: "latest"
+    hostname: "identity-mapper.sandbox.fynarfin.io"
```
Comment (fynmanoj) [inline helm/g2p-sandbox/values.yaml:500]: use mifos.io host name. fynarfin.io shoudl be used in g2p-sandbox-SIT chart

File: helm/g2p-sandbox/values.yaml [hunk 4]
```diff
@@ -490,3 +492,51 @@ ph-ee-engine:
...
+    ingress:
+      enabled: true
+      annotations:
+        kubernetes.io/ingress.class: nginx
+      tls:
+        - secretName: sandbox-secret
+      hosts:
+        - host: identity-mapper.sandbox.fynarfin.io
```
Comment (fynmanoj) [inline helm/g2p-sandbox/values.yaml:508]: use mifos.io host name. fynarfin.io shoudl be used in g2p-sandbox-SIT chart

File: helm/g2p-sandbox/values.yaml [hunk 5]
```diff
@@ -490,3 +492,51 @@ ph-ee-engine:
...
+
+    spring:
+      cache:
+        time_to_live: 30
+        time_to_idle: 30
+        max_entries_heap: 1000
+        max_byte_off_heap: 10
+        max_byte_disk: 10
+      datasource:
+        url: jdbc:mysql:thin://operationsmysql:3306/IdentityAccountMapper
```
Comment (fynmanoj) [inline helm/g2p-sandbox/values.yaml:540]: use lowercase db name

File: helm/identity-account-mapper/values.yaml [hunk 6]
```diff
@@ -171,34 +171,51 @@ kafka:
...
-      cpu: "1000m"
-      memory: "2Gi"
-    requests:
-      cpu: "200m"
-      memory: "0.5Gi"
+      kubernetes.io/ingress.class: nginx
+    tls:
+      - secretName: sandbox-secret
+    hosts:
+      - host: identity-mapper.sandbox.fynarfin.io
```
Comment (fynmanoj) [inline helm/identity-account-mapper/values.yaml:186]: use mifos.io

File: helm/identity-account-mapper/values.yaml [hunk 7]
```diff
@@ -171,34 +171,51 @@ kafka:
...
+
+  spring:
+    cache:
+      time_to_live: 30
+      time_to_idle: 30
+      max_entries_heap: 1000
+      max_byte_off_heap: 10
+      max_byte_disk: 10
+    datasource:
+      url: jdbc:mysql:thin://operationsmysql:3306/IdentityAccountMapper
```
Comment (fynmanoj) [inline helm/identity-account-mapper/values.yaml:218]: use lowercase db name

File: helm/identity-account-mapper/values.yaml [hunk 8]
```diff
@@ -171,34 +171,51 @@ kafka:
...
+                name: ph-ee-identity-account-mapper
+                port:
+                  number: 80
+  deployment:
+    annotations:
+      deployTime: "{{ .Values.deployTime }}"
+
+  logging:
+    level:
+      ROOT: DEBUG
```
Comment (fynmanoj) [inline helm/identity-account-mapper/values.yaml:200]: by default keep it as info

File: helm/identity-account-mapper/account-mapper/values.yaml [hunk 9]
```diff
@@ -30,32 +25,37 @@ async:
...
+spring:
+  cache:
+    time_to_live: 30
+    time_to_idle: 30
+    max_entries_heap: 1000
+    max_byte_off_heap: 10
+    max_byte_disk: 10
+  datasource:
+    url: jdbc:mysql:thin://operationsmysql:3306/IdentityAccountMapper
+    username: mifos
```
Comment (fynmanoj) [inline helm/identity-account-mapper/account-mapper/values.yaml:38]: check how these values are used ( ideally secret should be used)

File: helm/identity-account-mapper/account-mapper/values.yaml [hunk 10]
```diff
@@ -8,20 +8,15 @@ configmap:
   apiVersion: "v1"
 
 enabled: true
-replicas: 1
 image: "419830066942.dkr.ecr.ap-south-1.amazonaws.com/ph-ee-identity-account-mapper"
```
Comment (fynmanoj) [inline helm/identity-account-mapper/account-mapper/values.yaml:11]: point it to dockerhub

File: helm/identity-account-mapper/account-mapper/templates/role.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.managedServiceAccount }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: ph-ee-identity-account-mapper-role
```
Comment (fynmanoj) [inline helm/identity-account-mapper/account-mapper/templates/role.yaml:5]: no need to create roles for each service, we are using one role for all servcies now

File: helm/identity-account-mapper/values.yaml [hunk 12]
```diff
@@ -171,34 +171,51 @@ kafka:
...
       deployTime: "{{ .Values.deployTime }}"
 
-identity_account_mapper:
+account_mapper:
   enabled: true
-  image: "419830066942.dkr.ecr.ap-south-1.amazonaws.com/ph-ee-identity-account-mapper"
-  imagetag: "latest"
-  imagePullPolicy: "Always"
-  hostname: "identity.sandbox.fynarfin.io"
+  image: "419830066942.dkr.ecr.ap-south-1.amazonaws.com/ph-ee-vouchers"
```
Comment (fynmanoj) [inline helm/identity-account-mapper/values.yaml:176]: use account mapper image
use the tag inline

File: helm/identity-account-mapper/values.yaml [hunk 13]
```diff
@@ -171,34 +171,51 @@ kafka:
...
+  imageTag: "latest"
```
Comment (fynmanoj) [inline helm/identity-account-mapper/values.yaml:177]: you have removed this from the deployment already

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Comment by fynmanoj** [comment]
Please attach a ticker or story to each PR

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: helm/identity-account-mapper/account-mapper/templates/rolebinding.yaml [hunk 14]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.managedServiceAccount }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: ph-ee-operator-c-role-binding
```
[the remark this answers is not in the document]
Comment (apurbraj) [reply helm/identity-account-mapper/account-mapper/templates/rolebinding.yaml]: changes it to ph-ee-operator-role-binding

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**